redis/src/caching/cache_manager.rs

```diff
@@ -0,0 +1,226 @@
+use crate::caching::statistics::Statistics;
+use crate::caching::{CacheConfig, CacheStatistics};
+use crate::cmd::{CommandCacheInformation, CommandCacheInformationByRef};
+use crate::Value;
+use lru::LruCache;
+use std::cmp::min;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::Hasher;
+use std::num::NonZeroUsize;
+use std::ops::Add;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+use tokio::sync::watch::{channel, Receiver, Sender};
+
+type RedisCmd = Vec<u8>;
+type RedisKey = Vec<u8>;
+type LRUCacheShard = LruCache<RedisKey, CacheItem>;
+
+struct CacheCmdEntry {
+    cmd: RedisCmd,
+    value: Option<Value>,
+    receiver: Receiver<Value>,
+}
+
+struct CacheItem {
+    ttl: Instant,
```

Comment:
nit: time to live is a duration. if this is an instant, it's `expiry_time`, or some similar name.